Tools/machines/karolina-it4i/install_dependencies.sh

```diff
@@ -0,0 +1,47 @@
+#!/bin/bash
+#
+# Copyright 2023 The WarpX Community
+#
+# This file is part of WarpX.
+#
+# Author: Axel Huebl, Andrei Berceanu
+# License: BSD-3-Clause-LBNL
+
+# Exit on first error encountered #################################
+#
+set -eu -o pipefail
+
+# Check: ##########################################################
+#
+# Was karolina_warpx.profile sourced and configured correctly?
+if [ -z ${proj-} ]; then
+    echo "WARNING: The 'proj' variable is not yet set in your karolina_warpx.profile file!"
+    echo "Please edit its line 2 to continue!"
+    return
+fi
+
+# download and activate spack
+# this might take about ~ 1 hour
+if [ ! -d "$WORK/spack" ]
+then
+    git clone -c feature.manyFiles=true -b v0.21.0 https://github.com/spack/spack.git $WORK/spack
+    source $WORK/spack/share/spack/setup-env.sh
+
+    # create and activate the spack environment
+    spack env create warpx-karolina-cuda $WORK/src/warpx/Tools/machines/karolina-it4i/spack-karolina-cuda.yaml
+    spack env activate warpx-karolina-cuda
+    spack install
```

Comment:
```suggestion
    spack install
fi
```

I would also add an `else` branch that checks out the `v0.21.0` branch if the directory exists, removes the `warpx-karolina-cuda` environment, if it exists, and does the same install.

The idea about this script is, that you can run it multiple times and it will "refresh" your install if it should be needed, e.g., if you updated the docs and tell a user "please reinstall everything". In that case, you do not want to instruct how to "deinstall" the existing deps, because it gets confusing, instead automate it here.